app/src/lib/git/squash.ts

```diff
@@ -0,0 +1,69 @@
+import * as FSE from 'fs-extra'
+import { getCommits, revRange } from '.'
+import { Commit, CommitOneLine } from '../../models/commit'
+import { Repository } from '../../models/repository'
+import { getTempFilePath } from '../file-system'
+import { rebaseInteractive, RebaseResult } from './rebase'
+
+/**
+ * Squashes provided commits by calling interactive rebase
+ *
+ * @param toSquash - commits to squash onto another commit
+ * @param squashOnto  - commit to squash the `toSquash` commits onto
+ * @param lastRetainedCommitRef -  sha of commit before all commits in squash operation
+ * @param commitSummary - summary of new commit made by squash
+ * @param commitDescription - description of new commit made by squash
+ */
+export async function squash(
+  repository: Repository,
+  toSquash: ReadonlyArray<CommitOneLine>,
+  squashOnto: CommitOneLine,
+  lastRetainedCommitRef: string,
+  commitSummary: string,
+  commitDescription: string
+): Promise<RebaseResult> {
+  const commits: Commit[] = [
+    ...(await getCommits(repository, revRange(lastRetainedCommitRef, 'HEAD'))),
+  ]
+
+  let todoOutput = ''
+  commits.forEach(c => {
+    if (toSquash.map(sq => sq.sha).includes(c.sha)) {
+      return
+    }
+
+    if (c.sha === squashOnto.sha) {
```

Comment:
This should be pretty straightforward, but I think I'd add some comments above these ifs to explain a bit what/why it does what it does:
```suggestion
    // Ignore commits to squash because those are written right next to the target commit
    if (toSquash.map(sq => sq.sha).includes(c.sha)) {
      return
    }

    // If it's the target commit, write a `pick` command for it, and immediately after a `squash` line for every commit to squash
    if (c.sha === squashOnto.sha) {
```

What do you think? 😳